commands/fetch_templates_test.go

```diff
@@ -9,16 +9,46 @@ import (
 	"testing"
 )
 
-var SmallestZipFile = []byte{80, 75, 05, 06, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00}
-
 func Test_PullTemplates(t *testing.T) {
-	// Remove existing master.zip file if it exists
-	if _, err := os.Stat("master.zip"); err == nil {
-		t.Log("Found a master.zip file, removing it...")
+	defer tearDown_fetch_templates(t)
+
+	// Create fake server for testing.
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.ServeFile(w, r, "testdata/master_test.zip")
+	}))
+	defer ts.Close()
+
+	err := PullTemplates(ts.URL)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func Test_fetchTemplates(t *testing.T) {
+	defer tearDown_fetch_templates(t)
```

Comment:
This should be run at the start and at the end - because you may have had an inconsistent file-system if the files were written out and the code panicked and crashed mid-way.